### PR TITLE
fix: TableTransfer unselecting the current page is invalid

### LINKED ISSUE #48241: TableTransfer 反选当前页无效
### Reproduction link

[Edit on StackBlitz](https://stackblitz.com/run?file=demo.tsx)

### Steps to reproduce

// 请看ant官方提供的transferTable案例
// https://ant.design/components/transfer-cn#transfer-demo-table-transfer

### What is expected?

修复bug

### What is actually happening?

功能未按预期执行

| Environment | Info |
| --- | --- |
| antd | 5.16.0 |
| React | 18.2 |
| System | macOS:14.3.1 |
| Browser | chorm:122.0.6261.94 |

### PR DESCRIPTION
**Pull request by xiaozisong**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
#48241 

### 💡 Background and solution
Background：官网transfer组件demo中的table-transfer无法反选/取消反选
solution：list组件中第346行代码，在某些场景下组件是传递children了的，导致listBodyRef.current无法获取到keys数组，解决方法是通过判断listBodyRef.current.items的length，为空则取dataSource
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fix TableTransfer unselecting the current page is invalid      |
| 🇨🇳 Chinese |       修复TableTransfer 反选当前页无效    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/transfer/list.tsx [hunk 1]
```diff
@@ -343,7 +343,9 @@ const TransferList = <RecordType extends KeyWiseTransferItem>(
         label: selectInvert,
         onClick() {
           const availablePageItemKeys = getEnabledItemKeys(
-            (listBodyRef.current?.items || []).map((entity) => entity.item),
+            listBodyRef.current?.items?.length
+              ? listBodyRef.current?.items.map((entity) => entity.item)
```
Comment (yoyo837) [inline components/transfer/list.tsx:347]:
```suggestion
              ? listBodyRef.current.items.map((entity) => entity.item)
```

**Comment by xiaozisong** [comment]
> #48241
需要bug修复 刚才跑了一下 发现确实有这个问题

**Comment by yoyo837** [comment]
添加测试用例覆盖

**Comment by xiaozisong** [comment]
需要晚一点  等今晚或者明天

File: components/transfer/list.tsx [hunk 2]
```diff
@@ -345,8 +349,14 @@ const TransferList = <RecordType extends KeyWiseTransferItem>(
         key: 'selectInvert',
         label: selectInvert,
         onClick() {
+          const list = document.querySelector(`#${id} .ant-table-tbody`)?.childNodes || [];
```
Comment (afc163) [inline components/transfer/list.tsx:352]: 不应该有 dom 操作。

**Review by afc163 — CHANGES_REQUESTED** [review]
不应该有 dom 操作。

**Comment by afc163** [comment]
See https://github.com/ant-design/ant-design/issues/48241

**Pull request closed without merging**